cupy/fft/_callback.pyx

```diff
@@ -206,11 +207,18 @@ cdef inline str _prune(str temp_dir, str cache_dir, str _cufft_ver, str arch):
         if not os.path.isfile(cufft_lib_cached):
             p = subprocess.run([_nvprune, '-arch=sm_' + arch,
                                 cufft_lib_full, '-o', cufft_lib_temp],
-                               env=os.environ, cwd=temp_dir)
+                               env=os.environ, cwd=temp_dir,
+                               stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             p.check_returncode()
-            # atomic move with the destination guaranteed to be overwritten;
-            # using os.replace() is also ok here
-            os.rename(cufft_lib_temp, cufft_lib_cached)
+            if p.stderr:
+                # if no device code exists for sm_XX, nvprune exits normally
+                # with a warning printed to stderr
+                cufft_lib_pruned = None
+                _cufft_static_no_device_code.append(arch)
```

Comment:
Can we do an explicit check of the warning message here?

Reply:
Yes, let me add an assert.

Reply:
Added in cf6f1ab.